tests/test_diaCalculationPlugins.py

```diff
@@ -904,6 +912,152 @@ def testCalculate(self):
+        self.assertAlmostEqual(diaObjects.at[objId, "multiFap"],
+                               0.0)
+
+        # Test multi-band data with 2 sources.
+        n_sources = 2
+        times, fluxes = self.GeneratePeriodicData(n_sources, period=10)
+        diaObjects = pd.DataFrame({"diaObjectId": [objId]})
+        diaSources = pd.DataFrame(
+            data={"diaObjectId": n_sources * [objId],
+                  "band": n_sources * ["u"],
+                  "diaSourceId": np.arange(n_sources, dtype=int),
+                  "midpointMjdTai": times,
+                  "psfFlux": fluxes,
+                  "psfFluxErr": 1e-3+np.zeros(n_sources)})
+
+        plugin = LombScarglePeriodogramMulti(LombScarglePeriodogramMultiConfig(),
+                                             "ap_lombScarglePeriodogramMulti",
+                                             None)
+
+        run_multi_plugin(diaObjects, diaSources, "u",
+                         plugin)
+        self.assertTrue(np.isnan(diaObjects.at[objId, "multiPeriod"]))
+        self.assertTrue(np.isnan(diaObjects.at[objId, "multiPower"]))
+        self.assertTrue(np.isnan(diaObjects.at[objId, "multiFap"]))
+
+
+class TestLombScarglePeriodogram(unittest.TestCase):
+
+    def GeneratePeriodicData(self, n=10, period=10):
+        """Generate periodic data for testing Lomb Scargle Periodogram.
+
+        Parameters
+        ----------
+        n : int
+            Number of data points.
+        period : float
+            Period of the periodic signal.
+
+        Returns
+        -------
+        t : np.ndarray
+            Time values.
+        y_obs : np.ndarray
+            Observed flux values.
+        """
+        np.random.seed(42)
+
+        t = np.random.randint(100, size=n) + 0.3 + 0.4 * np.random.random(n)
+        y = 10 + np.sin(2 * np.pi * t / period)
+        dy = 0.001 + 0.001 * np.random.random(n)
+        y_obs = np.random.normal(y, dy)
+
+        return t, y_obs
+
+    def testCalculate(self):
+        """Test Lomb Scargle Periodogram."""
+        n_sources = 10
+        objId = 0
+
+        # Test period calculation.
+        times, fluxes = self.GeneratePeriodicData(n_sources, period=10)
+        diaObjects = pd.DataFrame({"diaObjectId": [objId]})
+        diaSources = pd.DataFrame(
+            data={"diaObjectId": n_sources * [objId],
+                  "band": n_sources * ["u"],
+                  "diaSourceId": np.arange(n_sources, dtype=int),
+                  "midpointMjdTai": times,
+                  "psfFlux": fluxes,
+                  "psfFluxErr": 1e-3+np.zeros(n_sources)})
+
+        plugin = LombScarglePeriodogram(LombScarglePeriodogramConfig(),
+                                        "ap_lombScarglePeriodogram",
+                                        None)
+
+        run_multi_plugin(diaObjects, diaSources, "u", plugin)
+        self.assertAlmostEqual(diaObjects.at[objId, "u_period"],
+                               9.994515)
```

Comment:
As above, where did this value come from? I think it should be exactly 10 (that's the period in your generated data), with `rtol=0.005`.

Reply:
Added appropriate comments